CONTRIBUTING.md

```diff
@@ -70,27 +70,97 @@ Sample curl request to invoke Ratify endpoint:
-1. build an image with your local changes
+There are some changes that should be validated in a cluster scenario.
+Follow the steps below to build and deploy a Ratify image with your private changes:
+
+#### build an image with your local changes
 ```
 docker build -f httpserver/Dockerfile -t yourregistry/deislabs/ratify:yourtag .
 ```
-2. [Authenticate](https://docs.docker.com/engine/reference/commandline/login/#usage) with your registry,  and push the newly built image
+
+#### [Authenticate](https://docs.docker.com/engine/reference/commandline/login/#usage) with your registry,  and push the newly built image
 ```
 docker push yourregistry/deislabs/ratify:yourtag
 ```
-3. Update [values.yaml](https://github.com/deislabs/ratify/blob/main/charts/ratify/values.yaml) to pull from your registry, when reusing image tag, setting pull policy to "Always" ensures we are pull the new changes
+
+#### Update [values.yaml](https://github.com/deislabs/ratify/blob/main/charts/ratify/values.yaml) to pull from your registry, when reusing image tag, setting pull policy to "Always" ensures we are pull the new changes
 ```json
 image:
   repository: yourregistry/deislabs/ratify
   tag: yourtag
   pullPolicy: Always
 ```
-4. Deploy from local helm chart
+
+### Deploy from local helm chart
+Deploy using one of the following deployments.
+
+**Option 1**
+Client and server auth disabled
 ```
 helm install ratify ./charts/ratify --atomic
 ```
+
+**Option 2**
+Client auth disabled and server auth enabled using self signed certificate
+
+1. Supply a certificate to use with Ratify (httpserver) or use the following script to create a self-signed certificate.
+
+```
+./scripts/generate-tls-cert.sh
+```
+
+2. Deploy using a certificate
+```
+helm install ratify ./charts/ratify \
+    --set provider.auth="tls" \
+    --set provider.tls.skipVerify=false \
+    --set provider.tls.cabundle="$(cat certs/ca.crt | base64 | tr -d '\n\r')" \
+    --set provider.tls.key="$(cat certs/tls.key)" \
+    --set provider.tls.crt="$(cat certs/tls.crt)" \
+    --atomic
+```
+
+**Option 3**
+Client auth disabled and server auth enabled using a secret
+
+*Note: There must be an existing secret in the 'default' namespace named 'ratify-cert-secret'.*
+
+```
+# Example secret schema
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ratify-cert-secret
+data:
+    tls.crt: <base64 crt value>
+    tls.key: <base64 key value>
+```
+
+*Note: The 'provider.tls.cabundle' must be supplied. Update the path or send in a base64 encoded value.*
+
+```
+helm install ratify ./charts/ratify \
+    --set provider.auth="tls" \
+    --set provider.tls.skipVerify=false \
+    --set provider.tls.cabundle="$(cat certs/ca.crt | base64 | tr -d '\n\r')" \
+    --atomic
+```
+
+**Option 4**
+Client / Server auth enabled (mTLS)  
+
+*Note: The cert must have a CN and subjectAltName name of ratify.gatekeeper-system*
+```
+helm install ratify ./charts/ratify \
+    --namespace gatekeeper-system
```

Comment:
might be worth adding a note that ratify must be installed in the same namespace as gatekeeper and that namespace must be updated to match gatekeeper's if the default gatekeeper installation namespace is not used

Reply:
Updated the note.